### ISSUE TITLE
Random hangs/lag on preset change

### ISSUE THREAD

**Issue by x2k13** [notspiff/visualization.projectm#66]
- Hardware: Fire Stick 4K
- Kodi version: 19.2 Matrix (armeabi-v7)
- projectM addon version: 19.0.0-Matrix

When using a preset directory containing loads of .milk presets (in my case 2000+), the random change function sometimes takes up to 3 seconds to switch to a different preset, making the audio stream (probably Kodi itself) hang/stop sound. This is weird, since the hardware should handle that easily, I assume. The presets directory is located directly on the internal memory of the Fire Stick 4K. 

I already opened an issue over at projectM's main repo here: https://github.com/projectM-visualizer/projectm/issues/545

It would be nice if someone could investigate this. Thank you.

**Comment by boondogglefob** [comment]
I can add hardware and software -  Nvidia Shield (P2987)  v8.2.3 Kodi 19.2 and 19.3

**Comment by x2k13** [comment]
> I can add hardware and software - Nvidia Shield (P2987) v8.2.3 Kodi 19.2 and 19.3

Meaning you have the same issues with Nvidia Shield and Kodi?

**Comment by boondogglefob** [comment]
Yes, that’s what I meant.  thank you.  I’ve played with more permutations of settings and options. I also did a vanilla install of 19.3. Seems to be something in specific preset packs. Only the Eyetune, the Milkdrop series, Mischa’s Collection and Stock preset packs seems to have the issue. Interestingly,  the ‘Stock” preset pack seems less predictable. 

The rest appear to be fine.  (bltc201, projectM, Tryptonaut, Yin, and Tests)

Hope this helps. Thanks again.

**Comment by ABerguerand** [comment]
Since then, it was identified that the issue occured only for the presets using shaders.
According to https://github.com/projectM-visualizer/projectm/issues/545 : 

> The Kodi plug-in is still using an old 3.1 version of libprojectM. They'll have to update the plug-in to use libprojectM 4 and ideally build against the soon-to-be-released 4.1 update, which will most existing issues and also provide way better visuals due to the vastly improved Milkdrop 2 compatibility.

Is this update to libprojectM 4 foreseen ?
